contracts/dnsregistrar/OffchainDNSResolver.sol

```diff
@@ -199,10 +200,24 @@ contract OffchainDNSResolver is IExtendedResolver, IERC165 {
         uint256 startIdx,
         uint256 lastIdx
     ) internal pure returns (bytes memory) {
-        // TODO: Concatenate multiple text fields
-        uint256 fieldLength = data.readUint8(startIdx);
-        assert(startIdx + fieldLength < lastIdx);
-        return data.substring(startIdx + 1, fieldLength);
+        uint256 totalLength = 0;
+        uint256 idx = startIdx;
+        while (idx < lastIdx) {
+            uint256 fieldLength = data.readUint8(idx);
+            totalLength += fieldLength;
+            idx += fieldLength + 1;
+        }
```

Comment:
```suggestion
        for (uint256 idx = startidx; idx < lastIdx; idx += fieldLength + 1) {
            uint256 fieldLength = data.readUint8(idx);
            totalLength += fieldLength;
        }
```

Reply:
made a slight change in the suggested code, since `fieldLength` variable initializes inside of the loop block;
https://github.com/ensdomains/ens-contracts/pull/332/commits/24fcb62a7f332f859e98e1e80d39770747f66792